### PR TITLE
Feat opponent

### PR DESCRIPTION
**Pull request by parkerpersaud**
https://trello.com/c/hYYde19q/21-425choose-an-opponent

### CODE REVIEW

File: src/main/java/me/karakelley/tictactoe/PlayerFactoryImplementation.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+public class PlayerFactoryImplementation implements PlayerFactory{
+  @Override
+  public Player makePlayer(String type, String marker) {
+    Player newPlayer = null;
+
+    if (type.equals("1")) {
+      newPlayer = new HumanPlayer(marker);
+    } else {
+      System.exit(0);
```
Comment (andarcabrera) [inline src/main/java/me/karakelley/tictactoe/PlayerFactoryImplementation.java]: The PlayerFactory has knowledge about when to exit the game. Couldn't this have been decided earlier? It seems like the minute you get an input that's not 1, you can exit the game, or throw some exception and catch it and then exit the game.

Also, at this point, you only have one type of player, the human player, but you already created a factory and are preparing for other type of players. As Aaron would say, don't overthink it. Do the minimum that needs to be done to meet the acceptance criteria for the given story.

**Review by andarcabrera — COMMENTED** [review]
Is the Trello board linked?

File: src/main/java/me/karakelley/tictactoe/GameConfiguration.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public Player getPlayer1() {
   }
 
   public void chooseOpponent(String choice) {
-    if (choice != "1") {
+    if (!choice.equals("1")) {
```
Comment (andarcabrera) [inline src/main/java/me/karakelley/tictactoe/GameConfiguration.java]: Magic strings are not your friends :). Extracting them into a constant is a better practice

Comment (parkerpersaud) [reply src/main/java/me/karakelley/tictactoe/GameConfiguration.java]: I'm not sure what you mean by magic strings. Can you clarify?

File: src/main/java/me/karakelley/tictactoe/PlayerTypeValidator.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  public boolean isValidInput(String input) {
+    try {
+      return inBounds(Integer.parseInt(input));
+    } catch(IllegalArgumentException e) {
+      return false;
+    }
+  }
+
+  private boolean inBounds(int choice) {
+    return choice > 0 && choice < 5;
```
Comment (aaronlahey) [inline src/main/java/me/karakelley/tictactoe/PlayerTypeValidator.java:14]: The logic for "what a valid choice is" is duplicated between this class and the factory. Can you think of a way to refactor?

**Pull request merged**